src/collection_notifications.hpp

```diff
@@ -88,10 +90,30 @@ struct CollectionChangeSet {
     // Per-column version of `modifications`
     std::vector<IndexSet> columns;
 
+    /// Partial sync states
+    realm::partial_sync::SubscriptionState partial_sync_old_state;
+    realm::partial_sync::SubscriptionState partial_sync_new_state;
+
+    // Any potential error message from partial sync for the new collection
+    // This is only set if partial_sync_new_state is ERROR
+    std::string partial_sync_error_message;
+
+    // Returns `true` if the modification lists are `empty`, either because
+    // no changes existed, or because they have have otherwise beeen
+    // consumed (By other parts of the Collection Notifier).
+    bool changes_moved() const noexcept {
```

Comment:
It's also not an accurate description of what the single assertion that uses it is asserting. The whole reason that there assert there is because the collection passed to `merge()` *isn't* simply moved from.

Reply:
Any suggestion on how to improve it then?

Reply:
Continue to assert that the changeset is empty after merge() is called and don't keep state around in strange places?